network/p2p/dht_test.go

```diff
@@ -146,6 +146,8 @@ func TestPubSubWithDHTDiscovery(t *testing.T) {
 	require.NoError(t, err)
 
 	for _, n := range nodes {
+		n := n
```

Comment:
undo?

Reply:
`n` is passed into the goroutine down the line so we need a copy of it.  It was always a bug, but it started racing more consistently with the new versions of libp2p libraries.

Reply:
you're passing s into the goroutine.  s != n

Reply:
n is used inside the function as:
```
ch <- n.Host().ID()
```

Reply:
nice catch!

From my perspective, it would be cleaner to avoid variable shadowing and write 
```golang
subReader := func(s *pubsub.Subscription, node *p2p.Node) {
  msg, err := s.Next(ctx)
  require.NoError(t, err)
  require.NotNil(t, msg)
  assert.Equal(t, data, msg.Data)
  ch <- node.Host().ID()
}
```
and then write in [line 165](https://github.com/onflow/flow-go/blob/b22e0a7c052b75b68659028e32c253fd2dd4cca3/network/p2p/dht_test.go#L165) 
```golang
go subReader(s, n)
```